go/vt/vttablet/tabletmanager/rpc_backup.go

```diff
@@ -100,6 +103,41 @@ func (tm *TabletManager) Backup(ctx context.Context, logger logutil.Logger, req
 			// Original type could be primary so pass in a real value for PrimaryTermStartTime
 			if err := tm.changeTypeLocked(bgCtx, originalType, DBActionNone, SemiSyncActionNone); err != nil {
 				l.Errorf("Failed to change tablet type from %v to %v, error: %v", topodatapb.TabletType_BACKUP, originalType, err)
+				return
+			}
+
+			// Find the correct primary tablet and set the replication source,
+			// since the primary could have changed while we executed the backup which can
+			// also affect whether we want to send semi sync acks or not.
+			tabletInfo, err := tm.TopoServer.GetTablet(bgCtx, tablet.Alias)
+			if err != nil {
+				l.Errorf("Failed to fetch updated tablet info, error: %v", err)
+				return
+			}
+
+			// Do not do anything for primary tablets and when active reparenting is disabled
```

Comment:
It's worth commenting why we do an early `return` when the tablet is `PRIMARY`, since this is the subject of this issue. I'd tell the story here, and point out that the replica may have been promoted as a `PRIMARY` even as the backup was running.